spec/API_specification/array_api/creation_functions.py

```diff
@@ -43,10 +43,11 @@ def asarray(obj: Union[array, bool, int, float, NestedSequence, SupportsBufferPr
            An object supporting the buffer protocol can be turned into a memoryview through ``memoryview(obj)``.
 
     dtype: Optional[dtype]
-        output array data type. If ``dtype`` is ``None``, the output array data type must be inferred from the data type(s) in ``obj``. If all input values are Python scalars, then
+        output array data type. If ``dtype`` is ``None``, the output array data type must be inferred from the data type(s) in ``obj``. If all input values are Python scalars, then, in order of precedence,
 
         -   if all values are of type ``bool``, the output data type must be ``bool``.
         -   if the values are a mixture of ``bool``\s and ``int``, the output data type must be the default integer data type.
```

Comment:
This one is a little odd, you can read it as explicitly disallowing `int`-only values.

Reply:
Will rephrase.

Reply:
This has now been addressed.